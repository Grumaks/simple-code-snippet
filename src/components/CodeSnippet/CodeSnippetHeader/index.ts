export { default as CodeSnippetHeader } from './CodeSnippetHeader';
