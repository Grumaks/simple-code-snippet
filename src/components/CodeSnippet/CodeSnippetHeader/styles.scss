$border: 1px solid #dcdcdc;

.code-snippet-header {
    background-color: #f7f7f7;
    border: $border;
    padding: 5px;
    display: flex;
    flex-direction: row;

    &-button {
        border-radius: 4px;
        border-width: 1px;
        cursor: pointer;
    }

    &-button + &-button {
        margin-left: 5px;
    }

    .right {
        margin-left: auto;
    }
}
