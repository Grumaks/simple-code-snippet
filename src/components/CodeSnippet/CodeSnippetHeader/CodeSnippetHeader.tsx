import * as React from 'react';

import './styles.scss';

export interface CodeSnippetHeaderProps {
    owner: string;
    repo: string;
    path: string;
    reference?: string
    showAll: () => void;
    copyAll: () => void;
    showButtons: boolean;
}

const CodeSnippetHeader = (props: CodeSnippetHeaderProps): JSX.Element => {

    const { owner, repo, path, reference, showAll, copyAll, showButtons } = props;

    // если в репозитории не master, а main, github сам сделает редирект
    const originalFileUrl =
        `https://github.com/${owner}/${repo}/blob/${reference ?? 'master'}/${path}`;
    const fileName = path.substr(path.lastIndexOf('/') + 1);

    return (
        <div className='code-snippet-header'>
            <a href={originalFileUrl} target='_blank' rel="noreferrer">{fileName}</a>
            { showButtons &&
                <>
                    <button className='code-snippet-header-button right' onClick={showAll} data-marker='showall'>Показать все</button>
                    <button className='code-snippet-header-button' onClick={copyAll}>Копировать все</button>
                </>
            }
        </div>
    );
};

export default React.memo(CodeSnippetHeader);

