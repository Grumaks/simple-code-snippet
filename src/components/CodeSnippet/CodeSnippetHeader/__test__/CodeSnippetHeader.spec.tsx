import * as React from'react';
import { mount } from 'enzyme';
import { CodeSnippetHeaderProps } from '../CodeSnippetHeader';
import { CodeSnippetHeader } from '..';

const showAllMock = jest.fn();
const copyAllMock = jest.fn();

const props: CodeSnippetHeaderProps = {
    owner: 'test',
    repo: 'test',
    path: 'test',
    reference: undefined,
    showAll: showAllMock,
    copyAll: copyAllMock,
    showButtons: true
};

describe('Test CodeSnippetHeader component', () => {

    const getComponent = (props: CodeSnippetHeaderProps) => {
        return <CodeSnippetHeader {...props}/>;
    };

    it('Should render correctly', () => {
        const wrapper = mount(
            getComponent(props)
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('Should call copyAll hadler when click on copyAll button', () => {
        const wrapper = mount(
            getComponent(props)
        );

        const button = wrapper.find('[data-marker="showall"]').first();
        button.simulate('click');

        expect(showAllMock).toHaveBeenCalledTimes(1);
    });
});
