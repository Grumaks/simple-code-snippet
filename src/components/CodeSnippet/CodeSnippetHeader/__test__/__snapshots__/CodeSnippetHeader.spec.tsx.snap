// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CodeSnippetHeader component Should render correctly 1`] = `
<Memo(CodeSnippetHeader)
  copyAll={[MockFunction]}
  owner="test"
  path="test"
  repo="test"
  showAll={[MockFunction]}
  showButtons={true}
>
  <div
    className="code-snippet-header"
  >
    <a
      href="https://github.com/test/test/blob/master/test"
      rel="noreferrer"
      target="_blank"
    >
      test
    </a>
    <button
      className="code-snippet-header-button right"
      data-marker="showall"
      onClick={[MockFunction]}
    >
      Показать все
    </button>
    <button
      className="code-snippet-header-button"
      onClick={[MockFunction]}
    >
      Копировать все
    </button>
  </div>
</Memo(CodeSnippetHeader)>
`;
